File: aula1003/C/limiteprocedural.c
// Linguagem C: O Colapso da Estrutura de Dados
struct Veiculo { int id; float combustivel; float km_rodados; };
struct Motorista { int id; int cnh_valida; };

// Acoplamento altíssimo: a função exige a injeção de múltiplos ponteiros
// Se adicionarmos a 'struct Clima', a assinatura da função quebra inteira.
void registrar_viagem(struct Veiculo *v, struct Motorista *m, float km) {
    if (m->cnh_valida == 1) {
        v->km_rodados = v->km_rodados + km;
        v->combustivel = v->combustivel - (km / 10.0);
    }
}

File: aula1003/python/limiteprocedural.py
# Estágio 3: O Colapso da Estrutura de Dados
veiculo = {'id': 1, 'combustivel': 50.0, 'km_rodados': 0.0}
motorista = {'id': 101, 'cnh_valida': True}

# Acoplamento altíssimo: a função exige a injeção de múltiplos dicionários
def registrar_viagem(v, m, km):
    if m['cnh_valida'] == True:
        v['km_rodados'] += km
        v['combustivel'] -= (km / 10.0)

registrar_viagem(veiculo, motorista, 100.0)